craft_parts/errors.py

```diff
@@ -482,15 +499,49 @@ class PluginBuildError(PartsError):
     :param plugin_name: The name of the plugin being processed.
     """
 
-    def __init__(self, *, part_name: str, plugin_name: str) -> None:
+    def __init__(
+        self, *, part_name: str, plugin_name: str, stderr: bytes | None = None
+    ) -> None:
         self.part_name = part_name
         self.plugin_name = plugin_name
+        self.stderr = stderr
         brief = f"Failed to run the build script for part {part_name!r}."
         resolution = f"Check the build output and verify the project can work with the {plugin_name!r} plugin."
         super().__init__(
             brief=brief, resolution=resolution, doc_slug="/reference/plugins.html"
         )
 
+    @property
+    @override
+    def details(self) -> str | None:
+        """Further details on the error.
+
+        Discards all trace lines that come before the last-executed script line
+        """
+        with contextlib.closing(StringIO()) as details_io:
+            if self.stderr is None:
+                return None
```

Comment:
I'd move this logic before you open `StringIO`, and fallback to the default implementation rather than returning None
```suggestion
        if self.stderr is None:
            return super().details
        with contextlib.closing(StringIO()) as details_io:
```